tests/integration/test_extreme_deduplication/test.py

```diff
@@ -49,20 +49,25 @@ def test_deduplication_window_in_seconds(started_cluster):
     node.query("INSERT INTO simple VALUES (0, 1)")
     assert TSV(node.query("SELECT count() FROM simple")) == TSV("2\n")
 
-    # wait clean thread
-    time.sleep(2)
+    # Wait for the cleanup thread.
```

Comment:
@alexey-milovidov haven't seen your PR (looked only issues) so I've submitted a fix as well - https://github.com/ClickHouse/ClickHouse/pull/52085

If you will prefer to merge yours, don't forget to adjust `max_cleanup_delay_period` setting.